piet-direct2d/src/text.rs

```diff
@@ -58,32 +81,25 @@ impl Text for D2DText {
     type TextLayout = D2DTextLayout;
 
     fn new_font_by_name(&mut self, name: &str, size: f64) -> Self::FontBuilder {
-        // Note: the name is cloned here, rather than applied using `with_family` for
-        // lifetime reasons. Maybe there's a better approach.
-        let builder = TextFormatBuilder::new(self.dwrite.clone()).size(size as f32);
-        D2DFontBuilder {
-            builder,
-            name: name.to_owned(),
-        }
+        let font = self
+            .dwrite
+            .system_font_collection()
+            .ok()
+            .and_then(|fonts| fonts.font_family(name))
+            .map(|family| D2DFont { family, size });
+        D2DFontBuilder { font }
     }
 
     fn system_font(&mut self, size: f64) -> Self::Font {
-        // this is... lazy. We try Segoe UI, and if it's missing we try Arial,
-        // and if that's missing we crash. :shrug_emoji:
-        // FIXME: something like this might be better? https://stackoverflow.com/questions/41505151/how-to-draw-text-with-the-default-ui-font-in-directwrite
-        D2DFont(
-            TextFormatBuilder::new(self.dwrite.clone())
-                .size(size as f32)
-                .family("Segoe UI".into())
-                .build()
-                .or_else(|_| {
-                    TextFormatBuilder::new(self.dwrite.clone())
-                        .size(size as f32)
-                        .family("Arial".into())
-                        .build()
-                })
-                .unwrap(),
-        )
+        let collection = self.dwrite.system_font_collection().unwrap();
+        //TODO: this is maybe not the best thing? I _think_ if we pass an empty string
```

Comment:
In practice I think we can expect "Segoe UI" to always succeed. At some point, we should test this on Windows 7.

Reply:
yea, i'm trying to think a bit more about what we should expect to exist on a given platform. If a user has deleted system fonts, I'm not sure we owe them much special treatment.